x-pack/plugins/apm/server/lib/service_map/get_service_map_from_trace_ids.test.ts

```diff
@@ -10,19 +10,106 @@ import { PromiseReturnType } from '../../../typings/common';
 import { fetchServicePathsFromTraceIds } from './fetch_service_paths_from_trace_ids';
 
 describe('getConnections', () => {
-  it('transforms a list of paths into a list of connections filtered by service.name and environment', () => {
-    const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
-      typeof fetchServicePathsFromTraceIds
-    >;
-    const serviceName = 'opbeans-node';
-    const environment = 'production';
-
-    const connections = getConnections(
-      response.aggregations?.service_map.value.paths,
-      serviceName,
-      environment
-    );
-
-    expect(connections).toMatchSnapshot();
+  describe('if no filter is given', () => {
+    it('includes all connections', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths
+      );
+
+      expect(connections).toMatchSnapshot();
+    });
+  });
+
+  describe('if service name and environment are given', () => {
+    it('excludes connections with different service name and environment', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const serviceName = 'opbeans-node';
+      const environment = 'production';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths,
+        serviceName,
+        environment
+      );
+      expect(
+        connections.every(
+          (conn) => conn.source['service.environment'] === environment
+        )
+      ).toBeTruthy();
+      expect(connections).toMatchSnapshot();
+    });
+  });
+
+  describe('if service name is given', () => {
+    it('excludes connections with different service name', () => {
+      const response = serviceMapFromTraceIdsScriptResponse as PromiseReturnType<
+        typeof fetchServicePathsFromTraceIds
+      >;
+      const serviceName = 'opbeans-node';
+
+      const connections = getConnections(
+        response.aggregations?.service_map.value.paths,
+        serviceName
+      );
+
+      expect(connections).toMatchSnapshot();
```

Comment:
Would it make sense to add:
```
      expect(
        connections.every(
          (conn) => conn.source['service.name'] === name
        )
      ).toBeTruthy();
```

?

Reply:
not really, even though I'm filtering by service name `opbeans-node` the return of the `getConnections` has other services too, which are connections from `opbeans-node`. For instance this is the list of service names returned:
```js
[ 'opbeans-python', 'opbeans-node', 'opbeans-ruby' ]
```
I can add an expect checking for this list instead of the snapshot though. WDYT?